app/controllers/api/v1/profiles_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Api::V1::ProfilesController < Api::V1::BaseController
+  before_action :authenticate_user!
+
+  def show
+    authorize(:profile, :show?)
+    render json: current_user
+  end
+
+  def update
+    authorize(:profile, :update?)
+    if current_user.update_attributes(avatar: params["avatar"])
```

Comment:
I highly recommend you to use `strong parameters` everywhere where it is possible.